NYPLAudiobookToolkit/Network/OpenAccessDownloadTask.swift

```diff
@@ -188,6 +202,53 @@ final class OpenAccessDownloadTask: DownloadTask {
         }
         return hash
     }
+  
+    // MARK: - Download Timer
+  
+    private func startDownloadTimer() {
+        serialQueue.async {
+            /// These value should be reset every time we start fetching,
+            /// so that we have the right time in case of retrying download
+            self.fetchStartTime = Date()
+            self.downloadTimeLimit = 30.0
+            
+            self.downloadTimer = NYPLRepeatingTimer(interval: .seconds(30),
+                                                    queue: self.serialQueue,
+                                                    handler: { [weak self] in
+                guard let self = self,
+                      let startTime = self.fetchStartTime else {
+                    return
+                }
+              
+                let elapsedTime = Date().timeIntervalSince(startTime)
+                if elapsedTime >= self.downloadTimeLimit {
+                    self.delegate?.downloadTaskExceededTimeLimit(self,
+                                                                 elapsedTime: elapsedTime)
+                    self.updateDownloadTimeLimit()
+                }
+            })
+        }
+    }
+  
+    fileprivate func removeDownloadTimer() {
+        serialQueue.async {
+            self.downloadTimer = nil
+        }
+    }
+  
+    /// We call the delegate when the download task has not completed
+    /// at the 30 seconds mark and 3 minutes (180 seconds) mark.
+    /// Therefore, we update the time limit every time the delegate is called and
+    /// remove the timer when we are done with it.
+    private func updateDownloadTimeLimit() {
+        serialQueue.async {
+            if self.downloadTimeLimit == 30.0 {
```

Comment:
all these hardcoded numbers are repeated for OverdriveDownloadTask, but they probably are meant to stay in sync. Should we move them maybe to the parent DownloadTask protocol as computed variables in a protocol extension? 

If these values are shared also with AudiobookNetworkService, we can use a Struct instead. But as far as I understand AudiobookNetworkService's constants are semantically different, so in that case they should not be linked.

Reply:
I am not sure how would logic work if it is a computed property. Can you explain on it?

Reply:
OverdriveDownloadTask and OpenAccessDownloadTask both inherit from DownloadTask. So we could have an extension like:
```
extension DownloadTask {
  static var timeoutValue: TimeInterval {
    return 660.0
  }
  // same for any constant that should be identical for all DownloadTasks
}
```
it has to be a computed property because protocol extensions do not allow to define static var/constants.

Reply:
Oh that's right, I misunderstood in the first place, thanks